exercise-3.sql

```diff
@@ -0,0 +1,107 @@
+select e.first_name,e.last_name,concat_ws(' ',m.first_name,m.last_name) as manager from employees e,employees m where e.manager_id = m.employee_id;
+
+-- 16)	write a SQL query to display the department name, city, and state province for each department.
+select d.department_name,l.city,l.state_province from departments d inner join locations l on d.location_id = l.location_id;
+
+-- 17) write a SQL query to list the employees (first_name , last_name, department_name) who belong to a department or don't
+select e.first_name,e.last_name,e.department_id,iff(e.department_id = null,'-',d.department_name) as belongs_too,iff(e.department_id = null,'-','Yes') as not_in_a_department from employees e left join departments d on e.department_id = d.department_id;
+
+-- 18) The HR decides to make an analysis of the employees working in every department. Help him to determine the salary given in average per department and the total number of employees working in a department.  List the above along with the department id, department name
+select e.department_id,d.department_name,round(avg(salary),2) as avg_salary,count(*) as number_of_employees from employees e inner join (select department_id,department_name from departments) as d on e.department_id = d.department_id group by e.department_id,d.department_name order by e.department_id;
+
+-- 19) Write a SQL query to combine each row of the employees with each row of the jobs to obtain a consolidated results. (i.e.) Obtain every possible combination of rows from the employees and the jobs relation.
+select * from employees cross join jobs;
+
+-- 20) Write a query to display first_name, last_name, and email of employees who are from Europe and Asia
+select e.first_name,e.last_name,e.email,c.country_name from employees e inner join departments d on e.department_id = d.department_id inner join locations l on d.location_id = l.location_id inner join countries c on l.country_id = c.country_id where lower(c.country_name) = 'canada' or lower(c.country_name) = 'germany';
+
+-- 21) Write a query to display full name with alias as FULL_NAME (Eg: first_name = 'John' and last_name='Henry' - full_name = "John Henry") who are from oxford city and their second last character of their last name is 'e' and are not from finance and shipping department.
+select concat_ws(' ',e.first_name, e.last_name) as full_name,d.department_name,l.city from employees e inner join departments d on e.department_id = d.department_id inner join locations l on d.location_id = l.location_id where lower(l.city) = 'seattle'and lower(e.last_name) like '%e_'and lower(d.department_name) in (select lower(department_name) from departments where lower(department_name) = 'finance' or lower(department_name) = 'shipping');
+
+-- 22) Display the first name and phone number of employees who have less than 50 months of experience
+select first_name,phone_number from employees where datediff('year',hire_date,current_date()) < 25;
+
```

Comment:
< 50

Reply:
We have no records less than 50 so I've changed it to 25. Should I change it back to 50?